File: backend-lamas/src/main/java/ch/lamas/backendlamas/trains/models/Ride.java
package ch.lamas.backendlamas.trains.models;

import java.sql.Date;
import java.sql.Timestamp;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "fahrt")
public class Ride {

    @Id
    int id;

    Date betriebstag;

    boolean ausfall;

    boolean zusatzfahrt;

    Timestamp ankunft;

    Timestamp ankunftprognose;

    PredictionStatus ankunftprognosestatus;

    Timestamp abfahrt;

    Timestamp abfahrtprognose;

    PredictionStatus abfahrtprognosestatus;

    boolean durchfahrt;

    boolean ankunftsverspaetung;

    boolean abfahrtsverspaetung;

    @ManyToOne()
    MeansOfTransportation verkehrsmittel;

    @ManyToOne()
    Route linien;

    @ManyToOne()
    Operator betreiber;

    @ManyToOne()
    Station haltestellen;
}

File: frontend-lamas/src/app/ride.ts
import { MeansOfTransportation } from "./Model/MeansOfTransportation";
import { Operator } from "./Model/Operator";
import { Route } from "./Model/Route";
import { Station } from "./Model/Station";
import { PredictionStatus } from "./predictionStatus";

export class Ride {
    id: string;
    betriebstag: Date;
    ausfall: boolean;
    zusatzfahrt: boolean;
    ankunft: Date;
    ankunftprognose: Date;
    ankunftprognosestatus: PredictionStatus;
    abfahrt: Date;
    abfahrtprognose: Date;
    abfahrtprognosestatus: PredictionStatus;
    durchfahrt: boolean;
    ankunftsverspaetung: boolean;
    abfahrtsverspaetung: boolean;
    verkehrsmittel: MeansOfTransportation;
    linien: Route;
    betreiber: Operator;
    haltestellen: Station;

    constructor(id, betriebstag, ausfall, 
        zusatzfahrt, ankunft, ankunftprognose, ankunftprognosestatus, 
        abfahrt, abfahrtprognose, abfahrtprognosestatus, durchfahrt,
        ankunftsverspaetung, abfahrtsverspaetung, verkehrsmittel,
        linien, betreiber, haltestellen) {

        this.id = id;
        this.betriebstag = betriebstag;
        this.ausfall = ausfall;
        this.zusatzfahrt = zusatzfahrt;
        this.ankunft = ankunft;
        this.ankunftprognose = ankunftprognose;
        this.ankunftprognosestatus = ankunftprognosestatus;
        this.abfahrt = abfahrt;
        this.abfahrtprognose = abfahrtprognose;
        this.abfahrtprognosestatus = abfahrtprognosestatus;
        this.durchfahrt = durchfahrt;
        this.ankunftsverspaetung = ankunftsverspaetung;
        this.abfahrtsverspaetung = abfahrtsverspaetung;
        this.verkehrsmittel = verkehrsmittel;
        this.linien = linien;
        this.betreiber = betreiber;
        this.haltestellen = haltestellen;
    }
}